### PR TITLE
Add inference serve example to run Stable Diffusion Inference using AWS Inferentia2

### LINKED ISSUE #43018: [Serve] Add inference serve example to run Stable Diffusion Inference using AWS Inferentia2
### Description

Add example of how to run inference using RayServe for Stable Diffusion model using AWS Inferentia2.
The example will showcase how to leverage AWS Inf2 instances to set up neuron libraries and use RayServe to run inference on a neuron compatible Stable Diffusion model. 

### Link

Add example for Stable Diffusion based inference similar to [aws-neuron BERT-base example](https://docs.ray.io/en/latest/serve/tutorials/aws-neuron-core-inference.html)

### PR DESCRIPTION
**Pull request by ratnopam**
## Why are these changes needed?
This example showcases how to serve Stable Diffusion Inference using AWS Inferentia2

## Related issue number

Closes #43018
## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Manual testing

Tested on a Inferentia(inf2.xl) instance (with 2 neuron_cores).

Serve deployment
```text
2024-02-07 17:53:28,299	INFO worker.py:1715 -- Started a local Ray instance. View the dashboard at http://127.0.0.1:8265 
(ProxyActor pid=25282) INFO 2024-02-07 17:53:31,751 proxy 172.31.10.188 proxy.py:1128 - Proxy actor fd464602af1e456162edf6f901000000 starting on node 5a8e0c24b22976f1f7672cc54f13ace25af3664a51429d8e332c0679.
(ProxyActor pid=25282) INFO 2024-02-07 17:53:31,755 proxy 172.31.10.188 proxy.py:1333 - Starting HTTP server on node: 5a8e0c24b22976f1f7672cc54f13ace25af3664a51429d8e332c0679 listening on port 8000
(ProxyActor pid=25282) INFO:     Started server process [25282]
(ServeController pid=25233) INFO 2024-02-07 17:53:31,921 controller 25233 deployment_state.py:1545 - Deploying new version of deployment StableDiffusionV2 in application 'default'. Setting initial target number of replicas to 1.
(ServeController pid=25233) INFO 2024-02-07 17:53:31,922 controller 25233 deployment_state.py:1545 - Deploying new version of deployment APIIngress in application 'default'. Setting initial target number of replicas to 1.
(ServeController pid=25233) INFO 2024-02-07 17:53:32,024 controller 25233 deployment_state.py:1829 - Adding 1 replica to deployment StableDiffusionV2 in application 'default'.
(ServeController pid=25233) INFO 2024-02-07 17:53:32,029 controller 25233 deployment_state.py:1829 - Adding 1 replica to deployment APIIngress in application 'default'.
Fetching 20 files: 100%|██████████| 20/20 [00:00<00:00, 195538.65it/s]
(ServeController pid=25233) WARNING 2024-02-07 17:54:02,114 controller 25233 deployment_state.py:2171 - Deployment 'StableDiffusionV2' in application 'default' has 1 replicas that have taken more than 30s to initialize. This may be caused by a slow __init__ or reconfigure method.
(ServeController pid=25233) WARNING 2024-02-07 17:54:32,170 controller 25233 deployment_state.py:2171 - Deployment 'StableDiffusionV2' in application 'default' has 1 replicas that have taken more than 30s to initialize. This may be caused by a slow __init__ or reconfigure method.
(ServeController pid=25233) WARNING 2024-02-07 17:55:02,344 controller 25233 deployment_state.py:2171 - Deployment 'StableDiffusionV2' in application 'default' has 1 replicas that have taken more than 30s to initialize. This may be caused by a slow __init__ or reconfigure method.
(ServeController pid=25233) WARNING 2024-02-07 17:55:32,418 controller 25233 deployment_state.py:2171 - Deployment 'StableDiffusionV2' in application 'default' has 1 replicas that have taken more than 30s to initialize. This may be caused by a slow __init__ or reconfigure method.
2024-02-07 17:55:46,263	SUCC scripts.py:483 -- Deployed Serve app successfully.
```

Sample Test
```
import requests
prompt = "a zebra is dancing in the grass, river, sunlit"
input = "%20".join(prompt.split(" "))
resp = requests.get(f"http://127.0.0.1:8000/imagine?prompt={input}")

print("Write the response to `output.png`.")
with open("output.png", "wb") as f:
    f.write(resp.content)
```

### CODE REVIEW

**Comment by ratnopamc** [comment]
This PR references an image that's submitted in https://github.com/ray-project/images/pull/18

File: doc/source/serve/doc_code/aws_neuron_core_inference_serve_stablediffusion.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+from io import BytesIO
```
[the remark this answers is not in the document]
Comment (ratnopamc) [reply doc/source/serve/doc_code/aws_neuron_core_inference_serve_stablediffusion.py]: Thanks for your review. Done

**Review by GeneDer — APPROVED** [review]
LGTM! Generated doc can be view https://anyscale-ray--43046.com.readthedocs.build/en/43046/serve/tutorials/aws-neuron-core-inference-stable-diffusion.html

**Review by angelinalg — APPROVED** [review]
Please add this example to this file: https://anyscale-ray--43046.com.readthedocs.build/en/43046/serve/tutorials/index.html

**Comment by peytondmurray** [comment]
Hi all, to add this to the new example gallery that we are building we need the following information:

- Skill level (beginner, intermediate, advanced)
- Frameworks (pytorch, deepspeed, etc)
- Use case (see the use cases section on the primary sidebar on the left of this page: https://docs.ray.io/en/latest/ray-overview/examples.html)

Thank you in advance!

**Comment by ratnopamc** [comment]
Generated doc can be viewed at https://anyscale-ray--43046.com.readthedocs.build/en/43046/serve/tutorials/index.html.

**Comment by GeneDer** [comment]
@edoakes can you help to merge this

**Comment by edoakes** [comment]
There's a merge conflict here

**Comment by GeneDer** [comment]
@ratnopamc can you address the merge conflicts as well? 🙏

File: doc/BUILD [hunk 2]
```diff
@@ -157,6 +158,7 @@ py_test_run_all_subdirectory(
     ],
     exclude = [
         "source/serve/doc_code/aws_neuron_core_inference_serve.py",
+        "source/serve/doc_code/aws_neuron_core_inference_serve_stable_diffusion.py"
```
Comment (GeneDer) [inline doc/BUILD:161]: @ratnopamc just fyi, missing a comma at the end here

Comment (ratnopamc) [reply doc/BUILD]: @GeneDer thanks so much for catching that. Fixed and resubmitted.

**Comment by GeneDer** [comment]
@edoakes this should be ready for merging now

**Pull request merged**